dashboard/modules/serve/serve_rest_api_impl.py

```diff
@@ -167,6 +167,16 @@ async def put_all_applications(self, req: Request) -> Response:
                     text=repr(e),
                 )
 
+            for deployment in [
+                d for app in config.applications for d in app.deployments
+            ]:
+                if "max_concurrent_queries" not in deployment.dict(exclude_unset=True):
+                    logger.warning(
+                        "The default value for `max_concurrent_queries` will "
+                        "change from 100 to 5 in an upcoming release."
+                    )
+                    break
```

Comment:
nit: put this in a helper method. in general it's much easier to read if things like this are pulled out of the main function body to avoid excessive nesting and length